sql/parser/sql.y

```diff
@@ -887,19 +890,25 @@ opt_boolean_or_string:
 
 // Timezone values can be:
 // - a string such as 'pst8pdt'
-// - an identifier such as "pst8pdt"
-// - an integer or floating point number
 // - a time interval per SQL99
-// name gives reduce/reduce errors against const_interval and LOCAL, so use
-// IDENT (meaning we reject anything that is a key word).
 zone_value:
-  SCONST { unimplemented() }
-| IDENT { unimplemented() }
-| const_interval SCONST opt_interval { unimplemented() }
-| const_interval '(' ICONST ')' SCONST { unimplemented() }
-| numeric_only { unimplemented() }
-| DEFAULT { unimplemented() }
-| LOCAL { unimplemented() }
+  SCONST
+  {
+    $$ = DString($1)
+  }
+| const_interval SCONST opt_interval
+  {
+    // TODO(pmattis): support opt_interval?
+    $$ = &CastExpr{Expr: DString($2), Type: $1}
```

Comment:
No calls to `EvalExpr`, but DEFAULT expressions are checked to see if they contain any vars. The benefit (which is of average size) is that parsing errors are currently more descriptive than errors caught later due to the precise positional information that we lose after parsing.